File: ViewModels/GameViews/GetDetailsGameResponseView.cs
using System.Collections.Generic;
using BJ.ViewModels.EnumsViews;

namespace BJ.ViewModels.GameViews
{
    public class GetDetailsGameResponseView
    {
        public UserGetDetailsGameResponseView User { get; set; }
        public List<BotGetDetailsGameResponseViewItem> Bots { get; set; }
        public string Winner { get; set; }

        public GetDetailsGameResponseView()
        {
            Bots = new List<BotGetDetailsGameResponseViewItem>();
        }

    }

    public class UserGetDetailsGameResponseView
    {
        public string Name { get; set; }
        public int Points { get; set; }
        public List<StepUserGetDetailsGameResponseViewItem> Cards { get; set; }

        public UserGetDetailsGameResponseView()
        {
            Cards = new List<StepUserGetDetailsGameResponseViewItem>();
        }
    }

    public class StepUserGetDetailsGameResponseViewItem
    {
        public SuitTypeView Suit { get; set; }
        public RankTypeView Rank { get; set; }

    }

    public class BotGetDetailsGameResponseViewItem
    {
        public string Name { get; set; }
        public int Points { get; set; }
        public List<StepBotGetDetailsGameResponseViewItem> Cards { get; set; }

        public BotGetDetailsGameResponseViewItem()
        {
            Cards = new List<StepBotGetDetailsGameResponseViewItem>();
        }

    }

    public class StepBotGetDetailsGameResponseViewItem
    {
        public SuitTypeView Suit { get; set; }
        public RankTypeView Rank { get; set; }
    }



}

File: BJ.WEB/ClientApp/src/app/shared/models/game-views/get-details-game-response-view.ts
import { SuitTypeView } from '../enums-views/suit-type-view';
import { RankTypeView } from '../enums-views/rank-type-view';

export class GetDetailsGameResponseView
{
    public user: UserGetDetailsGameResponseView;
    public bots: Array<BotGetDetailsGameResponseViewItem>;
    public winner: string;
    /**
     *
     */
    constructor() {
      this.bots = [];
    }
  }
  
  export class UserGetDetailsGameResponseView
  {
    public name:string;
    public points: number;
    public cards: Array<StepUserGetDetailsGameResponseViewItem>;
    /**
     *
     */
    constructor() {
      this.cards = [];
    }

  }

  export class StepUserGetDetailsGameResponseViewItem
  {
    public suit: SuitTypeView;
    public rank: RankTypeView;
  }

  export class BotGetDetailsGameResponseViewItem
  {
    public name:string;
    public points: number;
    public cards: Array<StepBotGetDetailsGameResponseViewItem>;
    /**
     *
     */
    constructor() {
     this.cards = [];
    }
  }

  export class StepBotGetDetailsGameResponseViewItem
  {
    public suit: SuitTypeView;
    public rank: RankTypeView;
  }